src/btree/bt_handle.c

```diff
@@ -881,8 +882,20 @@ __btree_get_last_recno(WT_SESSION_IMPL *session)
         return (WT_NOTFOUND);
 
     page = next_walk->page;
-    btree->last_recno = page->type == WT_PAGE_COL_VAR ? __col_var_last_recno(next_walk) :
-                                                        __col_fix_last_recno(next_walk);
+    last_recno = page->type == WT_PAGE_COL_VAR ? __col_var_last_recno(next_walk) :
+                                                 __col_fix_last_recno(next_walk);
+
+    /*
+     * If we skipped reading the page into memory, it's possible that next_walk points to an empty
+     * instantiated page rather than real content. When this happens, it has no entries and the last
+     * record number will be out of bounds, i.e. zero.
+     *
+     * In this context, the page also can't have an insert (or append) list, so it's safe to simply
+     * take the last ref's starting record number as the last record number of the tree.
+     */
+    if (last_recno == WT_RECNO_OOB)
+        last_recno = next_walk->key.recno;
+    btree->last_recno = last_recno;
 
     return (__wt_page_release(session, next_walk, 0));
```

Comment:
Add a test would not be too hard? In python, we can add some data, evict all the data. Truncate the end of the btree, close the btree. Make sure the truncate is globally visible, open the btree again and verify last_recno is not out of bounds?

Reply:
I thought so too - the trouble is the "open the btree again" step. I spent a bunch of time faffing about with RTS + closing/reopening connections and sessions. But I couldn't seem to poke it in quite the right way to hit the same code path.

I can explore it further but time-boxing the test to a few days seemed acceptable when the WT-11525 failure config repros it (eventually).

Reply:
Does close and reopen the database work？

Reply:
No, I keep ending up with an empty tree.

Reply:
The one thing that was unusual about the format runs was that it was shutting down all application threads, closing the sessions, but leaving the connection open. Maybe that would help with the repro. That would leave dhandles open in the connection.

Reply:
Unfortunately I've tried that one already. That path doesn't reopen the btree, so the `last_recno` hangs around and (correctly) remains high rather than being calculated as zero.